src/main/src/main/java/org/geoserver/config/impl/GeoServerEnvironmentAwareFacadeWrapper.java

```diff
@@ -0,0 +1,237 @@
+ *
+ */
+public class GeoServerEnvironmentAwareFacadeWrapper implements GeoServerFacade {
+
+    /**
+     * data access object
+     */
+    GeoServerFacade wrapped;
+    
+    /**
+     * 
+     */
+    public GeoServerEnvironmentAwareFacadeWrapper(GeoServerFacade facade) {
+        this.wrapped = facade;
+    }
+
+    @Override
+    public GeoServer getGeoServer() {
+        return wrapped.getGeoServer();
+    }
+
+    @Override
+    public void setGeoServer(GeoServer geoServer) {
+        wrapped.setGeoServer(geoServer);
+    }
+
+    @Override
+    public GeoServerInfo getGlobal() {
+        return wrapped.getGlobal();
+    }
+
+    @Override
+    public void setGlobal(GeoServerInfo global) {
+        wrapped.setGlobal(global);
+    }
+
+    @Override
+    public void save(GeoServerInfo geoServer) {
+        wrapped.save(geoServer);
+    }
+
+    @Override
+    public SettingsInfo getSettings(WorkspaceInfo workspace) {
+        return wrapped.getSettings(workspace);
+    }
+
+    @Override
+    public void add(SettingsInfo settings) {
+        wrapped.add(settings);
+    }
+
+    @Override
+    public void save(SettingsInfo settings) {
+        wrapped.save(settings);
+    }
+
+    @Override
+    public void remove(SettingsInfo settings) {
+        wrapped.remove(settings);
+    }
+
+    @Override
+    public LoggingInfo getLogging() {
+        return wrapped.getLogging();
+    }
+
+    @Override
+    public void setLogging(LoggingInfo logging) {
+        wrapped.setLogging(logging);
+    }
+
+    @Override
+    public void save(LoggingInfo logging) {
+        wrapped.save(logging);
+    }
+
+    @Override
+    public void add(ServiceInfo service) {
+        wrapped.add(service);
+    }
+
+    @Override
+    public void remove(ServiceInfo service) {
+        wrapped.remove(service);
+    }
+
+    @Override
+    public void save(ServiceInfo service) {
+        wrapped.save(service);
+    }
+
+    @Override
+    public Collection<? extends ServiceInfo> getServices() {
+        Collection wrappedServices = new ArrayList();
+        Request request = Dispatcher.REQUEST.get();
+        if (request != null && GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
+            // It is an OWS request; the placeholders must be expanded
+            Iterator<? extends ServiceInfo> it = wrapped.getServices().iterator();
+            while(it.hasNext()) {
+                ServiceInfo service = it.next();
+                if (service != null) {
+                    wrappedServices.add(service.clone(true));
+                }
+            }
+        } else {
+            wrappedServices = wrapped.getServices();
+        }
+        
+        return wrappedServices;
+    }
+
+    @Override
+    public Collection<? extends ServiceInfo> getServices(WorkspaceInfo workspace) {
+        Collection wrappedServices = new ArrayList();
+        Request request = Dispatcher.REQUEST.get();
+        if (request != null && GeoServerEnvironment.ALLOW_ENV_PARAMETRIZATION) {
```

Comment:
Duplicates code in the above method. Maybe get the services from the wrapped entity, and then call a method that will filter them and expand their variables? (mind, jdbc config returns a immutable list, make sure you keep on filtering by creating a new list)